flowpipe/plug.py

```diff
@@ -145,8 +145,6 @@ def value(self):
     @value.setter
     def value(self, value):
         """Propagate the dirty state to all connected Plugs as well."""
-        if self._sub_plugs:
```

Comment:
Does it make sense to check the the value that is set here is a dict-like structure with sub-plug names as keys? Or is that over-engineering at this point?

Reply:
Since the value is always re-compiled from the sub-plugs when accessing it, having a check here is not crucial I think. I'll add a separate ticket for this though, as having a warning at this point would be helpful for anyone who is unintentionally using it the wrong way